test/bases/renku_data_services/data_api/test_projects.py

```diff
@@ -153,6 +153,27 @@ async def test_project_creation_with_invalid_namespace(sanic_client, user_header
     assert "you do not have sufficient permissions" in response.json["error"]["message"]
 
 
+@pytest.mark.asyncio
+async def test_project_creation_with_conflicting_slug(sanic_client, user_headers, regular_user) -> None:
+    namespace = f"{regular_user.first_name}.{regular_user.last_name}"
+    payload = {
+        "name": "Existing project",
+        "namespace": namespace,
+        "slug": "my-project",
+    }
+    _, response = await sanic_client.post("/api/data/projects", headers=user_headers, json=payload)
+    assert response.status_code == 201, response.text
+
+    payload = {
+        "name": "Conflicting project",
+        "namespace": namespace,
+        "slug": "my-project",
+    }
+    _, response = await sanic_client.post("/api/data/projects", headers=user_headers, json=payload)
+
+    assert response.status_code == 409, response.text
```

Comment:
Note: prior to the fix, a HTTP 500 error would be returned, with leakage from postgres, see #308.